src/server/req_rerun.c

```diff
@@ -223,52 +223,30 @@ req_rerunjob(struct batch_request *preq)
 	int		  x, y, z;
 
 	jid = preq->rq_ind.rq_signal.rq_jid;
-	parent = chk_job_request(jid, preq, &jt);
-	if (parent == NULL)
+	if (FALSE == chk_job_request(jid, preq, &jt, &js, &pjob, &parent, NULL))
 		return;		/* note, req_reject already called */
 
 	if ((preq->rq_perm & (ATR_DFLAG_MGWR | ATR_DFLAG_OPWR)) == 0) {
 		req_reject(PBSE_PERM, 0, preq);
 		return;
 	}
 
-	if (jt == IS_ARRAY_NO) {
-
-		/* just a regular job, pass it on down the line and be done */
-
-		req_rerunjob2(preq, parent);
+	if ((pjob == NULL) && (jt == IS_ARRAY_Single)) {
+		req_reject(PBSE_BADSTATE, 0, preq);
 		return;
+	}
 
-	} else if (jt == IS_ARRAY_Single) {
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {
 
-		/* single subjob, if running can signal */
+		/* just a regular job or a subjob, pass it on down the line and be done */
 
-		offset = subjob_index_to_offset(parent, get_index_from_jid(jid));
-		if (offset == -1) {
-			req_reject(PBSE_UNKJOBID, 0, preq);
-			return;
-		}
-		i = get_subjob_state(parent, offset);
-		if (i == -1) {
-			req_reject(PBSE_IVALREQ, 0, preq);
-			return;
-		} else if (i == JOB_STATE_RUNNING) {
-			pjob = find_job(jid);		/* get ptr to the subjob */
-			if (pjob) {
-				req_rerunjob2(preq, pjob);
-			} else {
-				req_reject(PBSE_BADSTATE, 0, preq);
-				return;
-			}
-		} else {
-			req_reject(PBSE_BADSTATE, 0, preq);
-			return;
-		}
+		req_rerunjob2(preq, pjob);
```

Comment:
If a subjob is not in JOB_STATE_RUNNING then it used to be rejected earlier. Is this behavior changing now ?